app/javascript/components/record-form/form/subforms/subform-dialog-fields/component.jsx

```diff
@@ -64,8 +65,30 @@ const Component = ({
       }
     }
   }, []);
-
   return fieldsToDisplay.map(subformSectionField => {
+    let calculation_expression = subformSectionField?.calculation?.expression
+    if (calculation_expression) {
+      let count = Object.keys(values).reduce((prev, curr) => {
+        if (!(calculation_expression.sum.indexOf(curr) < 0) && values[curr] !== "") {
+          return prev + 1;
+        }
+
+        return prev;
+      }, 0);
+
+      if (count === 0) {
+        count = 1;
+      }
+
+      const calculatedVal = parseExpression(calculation_expression).evaluate(values);
+
+      const scoreCalField = `${formSection.unique_id.split("subform_section")[0]}score_calc`;
+      const scoreField = `${formSection.unique_id.split("subform_section")[0]}score`;
+
+      values[scoreCalField] = Math.floor(calculatedVal);
+      values[scoreField] = Math.floor(calculatedVal / count);
```

Comment:
I think this should be an `average` operator, so that we won't need to have the `count` logic here. On the other hand, I think you need to use `setFieldValue` to set the value in `formik`.